glean-core/src/metrics/mod.rs

```diff
@@ -81,6 +85,7 @@ impl Metric {
             Metric::Boolean(_) => "boolean",
             Metric::Counter(_) => "counter",
             Metric::Datetime(_, _) => "datetime",
+            Metric::Experiment(_) => "experiments",
```

Comment:
If we don't need to ask the metric for its category further below, maybe we should make this panic, so we catch when this gets called erroneously.

Reply:
Mh, what do you mean here? That we should remove the entry from `as_json` and then panic if we trip on Experiments in this function?

Reply:
```suggestion
            Metric::Experiment(_) => panic!("Experiments should not be serialized through this"),
```

* bikeshed on that wording.

than wherever we serialize experiments, avoid calling `category()` on them